### PR TITLE
cast to resolve type error between union type

### PR DESCRIPTION
**Pull request by aruniverse**

### CODE REVIEW

File: packages/service/src/introspection/IntrospectionClient.ts [hunk 1]
```diff
@@ -79,10 +79,10 @@ export class IntrospectionClient {
   }
 
   private async validateToken(accessToken: string): Promise<IntrospectionResponse> {
-    const decoded = jwt.decode(accessToken, { complete: true });
+    const decoded = jwt.decode(accessToken, { complete: true, json: true });
     if (!decoded)
       throw new Error("Failed to decode JWT");
-    const { payload, header } = decoded;
+    const { payload, header } = decoded as { payload: jwt.JwtPayload, header: jwt.JwtHeader };
```
Comment (calebmshafer) [inline packages/service/src/introspection/IntrospectionClient.ts:85]: what was the exact issue before? Was `decode` not returning as an object with the payload and header props?

Comment (aruniverse) [reply packages/service/src/introspection/IntrospectionClient.ts:85]: for some reason ts couldnt tell whether payload was a string or a JwtPayload

Comment (aruniverse) [reply packages/service/src/introspection/IntrospectionClient.ts:85]: image

Comment (calebmshafer) [reply packages/service/src/introspection/IntrospectionClient.ts:85]: okay, just tested.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**